### PR TITLE
Adding staking to Notional DSA connector

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: contracts/mainnet/connectors/notional/events.sol [hunk 1]
```diff
@@ -2,90 +2,118 @@
 pragma solidity ^0.7.6;
 
 contract Events {
-    event LogDepositCollateral(
```
Comment (jeffywu) [inline contracts/mainnet/connectors/notional/events.sol]: @weitianjie2000 can we revert some of the formatting errors

**Review by jeffywu — CHANGES_REQUESTED** [review]

File: contracts/mainnet/connectors/notional/main.sol [hunk 2]
```diff
@@ -655,6 +655,139 @@ abstract contract NotionalResolver is Events, Helpers {
...
+		approve(weth, address(staking), wethAmount);
+
+		staking.mintFromWETH(noteAmount, wethAmount, minBPT);
+
+		_eventName = "LogMintSNoteFromWETH(address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(address(this), noteAmount, wethAmount, minBPT);
+	}
+
+	/// @notice Begins a cool down period for the sender, this is required to redeem tokens
+	/// @dev Current cool down period is set to 1296000 seconds (15 days)
```
Comment (jeffywu) [inline contracts/mainnet/connectors/notional/main.sol]: Maybe dont leave this comment in here, we might change it in the future.

File: contracts/mainnet/connectors/notional/main.sol [hunk 3]
```diff
@@ -655,6 +655,139 @@ abstract contract NotionalResolver is Events, Helpers {
...
+		bool redeemWETH
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		if (sNOTEAmount == type(uint256).max)
+			sNOTEAmount = staking.balanceOf(address(this));
+
+		approve(staking, address(staking), sNOTEAmount);
```
Comment (jeffywu) [inline contracts/mainnet/connectors/notional/main.sol]: Is this approval required? I don't think so right?

Comment (ghost) [reply contracts/mainnet/connectors/notional/main.sol]: remvoed

File: contracts/mainnet/connectors/notional/main.sol (hunk 2 above)
Comment (ghost) [reply contracts/mainnet/connectors/notional/main.sol]: removed

File: contracts/mainnet/connectors/notional/events.sol (hunk 1 above)
Comment (ghost) [reply contracts/mainnet/connectors/notional/events.sol]: done

**Pull request merged**